### ISSUE TITLE
一些疑问

### ISSUE THREAD

**Issue by LeighSu** [gentaiscool/end2end-asr-pytorch#13]
为了表达地更清晰些还是直接用中文，请勿见怪。
1. data_loader.py的136-137行，在读取transcript文件时为什么需要增加SOS_CHAR,EOS_CHAR?
with open(transcript_path, 'r', encoding='utf8') as transcript_file:
            transcript = constant.SOS_CHAR + transcript_file.read().replace('\n', '').lower() + constant.EOS_CHAR

**Comment by LeighSu** [comment]
2. train.py的49行 增加了PAD_CHAR的目的是什么呢？
labels = constant.PAD_CHAR + constant.SOS_CHAR + constant.EOS_CHAR + labels

**Comment by LeighSu** [comment]
3. 在计算train和eval的loss时，用的是model计算出来的gold作为groud truth为什么不是用tgt做的group truth?
cer = calculate_cer(strs_hyps[j].replace(' ', ''), strs_gold[j].replace(' ', ''))
wer = calculate_wer(strs_hyps[j], strs_gold[j])

**Comment by gentaiscool** [comment]
Let me translate your questions and answer them in English so everyone can follow.
1. Why do I need to use SOS_CHAR and EOS_CHAR?
> These characters are special characters that identify the start and end of transcripts.

2. Why do I need to use PAD_CHAR?
> The data used in the training can have different lengths and in tensor operations (such as, multiplication or addition), we need to have fixed-dimension tensors on each batch. 

3. When calculating the loss of the train and eval, why don't you use the target as the ground truth?
> I didn't get this question. But, I can say that I calculate CER and WER by comparing the string generated by the model to the ground truth (gold).
